### PR TITLE
ssocr: unstable-2018-08-11 -> 2.22.1

### PR DESCRIPTION
**Pull request by ghost**
ZHF: #172160

###### Description of changes

Prior to this commit, ssocr's build was broken due to imlib moving to
pkg-config and removing the imlib2-config program.

This commit upgrades to the latest upstream, which understands
pkg-config.  It also corrects some buildInputs/nativeBuildInputs
hygiene issues.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by a-m-joseph** [comment]
The ZHF announcement PR says: "Please ping @NixOS/nixos-release-managers on the PR and add the 0.kind: build failure label to the pull request. If you're unable to because you're not a member of the NixOS org please ping @dasJ, @tomberek, @jonringer, @Mic92"

I am unable to add this build label, so I therefore am issuing this ping.

File: pkgs/applications/misc/ssocr/default.nix [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
     owner = "auerswal";
     repo = "ssocr";
-    rev = "5e47e26b125a1a13bc79de93a5e87dd0b51354ca";
-    sha256 = "0yzprwflky9a7zxa3zic7gvdwqg0zy49zvrqkdxng2k1ng78k3s7";
+    rev = "629ecd05f65f9a48d8330100380c30b5fd9147e4";
+    sha256 = "sha256-j1l1o1wtVQo+G9HfXZ1sJQ8amsUQhuYxFguWFQoRe/s=";
   };
 
-  nativeBuildInputs = [ imlib2 libX11 ];
+  nativeBuildInputs = [ imlib2 pkg-config ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ssocr/default.nix:14]:
```suggestion
  nativeBuildInputs = [ pkg-config ];
```

File: pkgs/applications/misc/ssocr/default.nix [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
   pname = "ssocr";
-  version = "unstable-2018-08-11";
+  version = "2.22.1";
 
   src = fetchFromGitHub {
     owner = "auerswal";
     repo = "ssocr";
-    rev = "5e47e26b125a1a13bc79de93a5e87dd0b51354ca";
-    sha256 = "0yzprwflky9a7zxa3zic7gvdwqg0zy49zvrqkdxng2k1ng78k3s7";
+    rev = "629ecd05f65f9a48d8330100380c30b5fd9147e4";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ssocr/default.nix:10]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/misc/ssocr/default.nix [hunk 3]
```diff
@@ -1,17 +1,18 @@
-{ lib, stdenv, fetchFromGitHub, imlib2, libX11 }:
+{ lib, stdenv, fetchFromGitHub, imlib2, libX11, pkg-config }:
 
 stdenv.mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ssocr/default.nix:3]:
```suggestion
stdenv.mkDerivation rec {
```

**Comment by a-m-joseph** [comment]
Tested and squashed.  Thanks!

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 172444` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ssocr</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**